src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.hpp

```diff
@@ -952,7 +953,7 @@ class ShenandoahScanRemembered: public CHeapObj<mtGC> {
   void coalesce_objects(HeapWord *addr, size_t length_in_words);
 
   // Return true iff this object is "properly" registered.
-  bool verify_registration(HeapWord* address, size_t size_in_words);
+  bool verify_registration(HeapWord* address, ShenandoahMarkingContext* ctx);
```

Comment:
There's some trickery with the value of the ctx argument that should probably be described in the API comment.  In particular, ctx is non-null iff the the marking context is "valid" for the purposes of finding live objects within old-gen memory.  It is not valid if we are doing a young collection while old-gen is concurrently marking.  When ctx is NULL, it is required that the old-gen memory has been coalesced and filled.  During mixed evacuations, we rely upon ctx within old-gen because we know that a new old-gen concurrent mark cannot begin until the previously selected evacuation candidates have all been collected.  We don't bother to coalesce-and-fill the collection candidates, but they may be scanned multiple times by young-gen (mixed) remembered set scanning while we are waiting for these candidate regions to be collected.  It's not clear to me that this patch honors all of these behaviors.

Reply:
This method is called from the other rset verify methods (init-mark, update-references, after-full-gc). Here I've just made it take the `ctx` computed in the caller (rather than re-selecting it in the called method). The size parameter was unused.